lib/apphook.c

```diff
@@ -82,7 +82,7 @@ register_application_hook(gint type, ApplicationHookFunc func, gpointer user_dat
       entry->func = func;
       entry->user_data = user_data;
 
-      application_hooks = g_list_append(application_hooks, entry);
+      application_hooks = g_list_prepend(application_hooks, entry);
```

Comment:
hmmm... I need to check code (on my computer, currently I have only my phone), but are you sure that this kind of reordering (or reverse ordering) is the best solution?

Reply:
I was thinking about:
1. "permanent" hooks, but the current hook is general enough if there is a correct option to re-register (did not want to extend interface if possible)
2. move hooks that is going to be triggered into some tmp list, and trigger them one-by-one (so two distinct list)
3. insert hooks before the current (this is a kinda variation of the `2.`)

Mainly I was thinking about `2.` and `3.`, possible the `3.` would be more readable and memory allocation/copy should not be a concern here. 

I do not mind the change of ordering, as there was no guarantee of order before; and imho should not be.

What do you think ?

Reply:
okay, thanks for detailed answer
(it would be nice to share these kind of thoughts next time :) )

Reply:
> I do not mind the change of ordering, as there was no guarantee of order before; and imho should not be.

Can you explain what do you mean by "no guarantee of order"? Weren't the hooks executed in the order they were registered?

Reply:
Before this patch they were, but not after. As the hooks for now are inserted into the list head.